File: Baekjoon/Greedy/15904.py
# https://www.acmicpc.net/problem/15904

import sys
input = sys.stdin.readline

s = input().strip()
ucpc = "UCPC"
idx = 0

for c in s:
    if c == ucpc[idx]:
        idx += 1
    if idx == 4:
        break
    
print("I love UCPC" if idx == 4 else "I hate UCPC")

File: Baekjoon/Greedy/15904.cpp
// https://www.acmicpc.net/problem/15904

#include <iostream>
#include <string>
using namespace std;

int main() {
    string s, ucpc = "UCPC";
    getline(cin, s);

    int idx = 0;
    for (char c : s) {
        if (c == ucpc[idx]) {
            idx++;
        }
        if (idx == 4) {
            break;
        }
    }

    if (idx == 4) {
        cout << "I love UCPC" << '\n';
    } else {
        cout << "I hate UCPC" << '\n';
    }

    return 0;
}